ckanext/qdes_schema/helpers.py

```diff
@@ -62,3 +63,141 @@ def qdes_relationship_types_choices(field):
+    get_action('update_related_resources')(context, data_dict)
+
+
+def create_series_or_collection_relationships(context, pkg_dict):
+    series_or_collection = pkg_dict.get('series_or_collection', [])
+    datasets = get_converter('json_or_string')(series_or_collection)
+    relationship_type = 'isPartOf'
+    add_related_resources(pkg_dict, datasets, relationship_type)
+
+
+def create_related_datasets_relationships(context, pkg_dict):
+    related_datasets = pkg_dict.get('related_datasets', [])
+    datasets = get_converter('json_or_string')(related_datasets)
+    relationship_type = 'unspecified relationship'
+    add_related_resources(pkg_dict, datasets, relationship_type)
+
+
+def add_related_resources(pkg_dict, datasets, relationship_type):
+    if not datasets or not isinstance(datasets, list):
+        return
+    related_resources = pkg_dict.get('related_resources', {})
+    related_resources = get_converter('json_or_string')(related_resources)
+    if not related_resources:
+        related_resources = {"resource": [], "relationship": [], "count": 0}
+
+    for dataset in datasets:
+        if not any(resource for resource in related_resources.get("resource", []) if resource == dataset):
+            related_resources["resource"].append(dataset)
+            related_resources["relationship"].append(relationship_type)
+            related_resources["count"] += 1
+
+    pkg_dict['related_resources'] = related_resources
+
+
+def create_related_resource_relationships(context, pkg_dict):
+    related_resources = pkg_dict.get('related_resources', {})
+    related_resources = get_converter('json_or_string')(related_resources)
+    if related_resources and isinstance(related_resources, dict):
+        dataset_id = pkg_dict.get('id')
+        resources = related_resources.get('resource', [])
+        relationships = related_resources.get('relationship', [])
+
+        create_relationships(context, dataset_id, resources, relationships)
+
+
+def create_relationships(context, dataset_id, datasets, relationships):
+    try:
+        for index, dataset in enumerate(datasets):
+            relationship_type = relationships[index]
+            relationship_dataset, relationship_url = get_dataset_relationship(context, dataset)
+
+            if relationship_dataset or relationship_url:
+                relationship = get_action('package_relationship_create')(context, {
+                    'subject': dataset_id,
+                    'object': relationship_dataset,
+                    'type': relationship_type,
+                    'comment': relationship_url,
+                })
+    except Exception as e:
+        log.debug('create_relationships error: {0}'.format(e))
+        raise
+
+
+def get_dataset_relationship(context, dataset):
+    relationship_dataset = None
+    relationship_url = None
+    try:
+        get_validator('package_id_exists')(dataset, context)
```

Comment:
@MarkCalvert 

I changed this to use the `package_id_exists` validator instead of `package_name_exists` validator to remedy the issue where the ID of the dataset was displaying in the dataset details page, e.g.

![image](https://user-images.githubusercontent.com/26393708/93321402-38b59b80-f866-11ea-8620-96234c4bea65.png)